greeter/README.md

```diff
@@ -1,21 +1,474 @@
+
+```ts
+/**
+ * Greeting message response
+ */
+export interface Message {
+  timestamp: Date;
+  language: string;
+  greeting: string;
+}
+```
+
+### Step 2: Create a Greeting Controller
+
+1. Create an endpoint `/greet/{name}` that takes a name and returns a greeting message in the language that is specified in the HTTP request. Run the Controller generator: 
+
+    ```sh
+    $ lb4 controller
+    ? Controller class name: Greeting
+    Controller Greeting will be created in src/controllers/greeting.controller.ts
+
+    ? What kind of controller would you like to generate? Empty Controller
+    create src/controllers/greeting.controller.ts
+    update src/controllers/index.ts
+
+    Controller Greeting was created in src/controllers/
+    ```
+
+2. In the generated GreetingController in `src/controllers/greeting.controller.ts`, modify the constructor:
+    ```ts
+    constructor(
+        @inject(GREETING_SERVICE) private greetingService: GreetingService,
+        @inject(RestBindings.Http.REQUEST) private request: Request,
+    ) {}
+    ```
+
+3. Add the endpoint `/greet/{name}` which calls the [GreetingService](https://github.com/strongloop/loopback-next/blob/master/examples/greeter-extension/src/greeting-service.ts) in the greeter-extension that we've created already. It accepts language English or Chinese. If other language is specified, it will fall back to English.
+
+    ```ts
+    @get('/greet/{name}', {
+    responses: {
+    '200': {
+        description: '',
+        content: {
+        'application/json': {
+            schema: {
+            type: 'object',
+            properties: {
+                timestamp: 'string',
+                language: 'string',
+                message: 'string',
+            },
+            },
+        },
+        },
+    },
+    },
+    })
+    async greet(
+        @param.path.string('name') name: string,
+        @param.header.string('language') lan: string,
+    ): Promise<Message> {
+        const language: string =
+        this.request.acceptsLanguages(['en', 'zh']) || 'en';
+        const greeting = await this.greetingService.greet(language, name);
+        return {
+        timestamp: new Date(),
+        language,
+        greeting,
+        };
+    }
+    ```
+4. Add the necessary imports
+
+    ```ts
+    import { param, get, Request, RestBindings } from "@loopback/rest";
+    import { Message } from "../types";
+    import {inject} from '@loopback/context';
+    import {
+        GreetingService,
+        GREETING_SERVICE,
+    } from '@loopback/example-greeter-extension';
+    ```
+
+5. Make sure we install the extension: 
+    ```sh
+    npm i --save @loopback/example-greeter-extension
+    ```
+
+### Step 3: Bind the GreetingService to the application
+
+Under `src/application.ts`, add the following line inside the constructor:
+```ts
+    this.component(RestExplorerComponent);
+
+    // add this line
```

Comment:
// and import `GreetingComponent` from `@loopback/example-greeter-extension` by
// import {GreetingComponent} from '@loopback/example-greeter-extension';